### DISCUSSION TITLE
Additional parameters for stream on demand

### DISCUSSION THREAD

**Discussion by msokolovich** [bluenviron/mediamtx#526] [Questions]
Aler9, thanks for your app. It's better, simpler and lighter than other apps and modules. 

I have one little question. 

I want to get HLS stream on demand with specify server, channel and subtype.
Is it possible to use variables for runOnDemand? Something like that
runOnDemand: ffmpeg -probesize 32 -analyzeduration 0 -i "rtsp://user:pass@**$SERVER**:554/cam/realmonitor?channel=**$CHANNEL**&subtype=**$SUBTYPE**" -c:v libx264 -vf fps=20 -f rtsp rtsp://localhost:$RTSP_PORT/**$SERVER**/**$CHANNEL**/**$SUBTYPE**

**Comment by msokolovich** [comment]
Someone can show me examples for this?

###############################################
# Path parameters

# these settings are path-dependent.
# it's possible to use regular expressions by using a tilde as prefix.
# for example, "~^(test1|test2)$" will match both "test1" and "test2".
# for example, "~^prefix" will match all paths that start with "prefix".
# the settings under the path "all" are applied to all paths that do not match
# another entry.

**Comment by aler9** [comment]
> Is it possible to use variables for runOnDemand? Something like that

Any environment variable passed to `rtsp-simple-server` will also be available to the custom commands:

```
MY_ENV=testing ./rtsp-simple-server
```

`MY_ENV` will be available to any subcommand.

> Someone can show me examples for this?

```yml
paths:
  '~^all_paths_that_starts_with_this':
    readUser: myuser
    readPass: mypass
```

[a comment by an automated account is not shown]
